src/Workspaces/VisualBasic/Portable/CodeGeneration/EventGenerator.vb

```diff
@@ -74,7 +75,64 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.CodeGeneration
         Private Function GenerateEventDeclarationWorker([event] As IEventSymbol,
                                                         destination As CodeGenerationDestination,
                                                         options As CodeGenerationOptions) As DeclarationStatementSyntax
-            ' TODO(cyrusn): Handle Add/Remove/Raise events
+
+            If options.GenerateMethodBodies AndAlso
+                ([event].AddMethod IsNot Nothing OrElse [event].RemoveMethod IsNot Nothing OrElse [event].RaiseMethod IsNot Nothing) Then
+                Return GenerateCustomEventDeclarationWorker([event], destination, options)
+            Else
+                Return GenerateNotCustomEventDeclarationWorker([event], destination, options)
+            End If
+        End Function
+
+        Private Function GenerateCustomEventDeclarationWorker(
+                [event] As IEventSymbol, 
+                destination As CodeGenerationDestination, 
+                options As CodeGenerationOptions) As DeclarationStatementSyntax
+            Dim addStatements = If(
+                [event].AddMethod Is Nothing, 
+                New SyntaxList(Of StatementSyntax), 
+                GenerateStatements([event].AddMethod))
+            Dim removeStatements = If(
```

Comment:
Personally I think this level of wrapping is a bit too much, especially for such short arguments. However I mostly see this style used when wrapping. Alternatively I could go for something like:
```
If([event].AddMethod Is Nothing,
    New SyntaxList(Of StatementSyntax), 
    GenerateStatements([event].AddMethod))
```